### PR TITLE
Deprecate legacy restore (prior to NuoDB 4.2)

### PR DESCRIPTION
**Pull request by sivanov-nuodb**
**Changes**

- Deprecate legacy restore machinery which was used prior to NuoDB 4.2.
- Remove undocumented `nuodb.requestKey`, `nuodb.latestPrefix` and `nuodb.latestKey` Helm values.

**Testing**

- removed the E2E test that exercises backwards compatibility with legacy restore
- regression testing

### CODE REVIEW

File: stable/database/files/nuosm [hunk 1]
```diff
@@ -508,311 +437,85 @@ log "archiveId=$myArchive; DB=$DB_NAME; hostname=$HOSTNAME"
 atomCount=$( ls -l $DB_DIR/*.atm 2>/dev/null | wc -l )
 catalogCount=$( ls -l $DB_DIR/*.cat 2>/dev/null | wc -l )
 
-if isRestoreRequestSupported && [ -z "$legacy_restore_requested" ]; then
```
Comment (sivanov-nuodb) [inline stable/database/files/nuosm:511]: Feel free to skim the rest of this file. The following changes are done:
- The code in the _if_ block just has an indentation change.
- The code in the _else_ block is removed.

File: stable/restore/files/nuorestore [hunk 2]
```diff
@@ -181,102 +168,66 @@ if ! isUrl "$restore_source"; then
    fi
 fi
 
-if isRestoreRequestSupported; then
```
Comment (sivanov-nuodb) [inline stable/restore/files/nuorestore:184]: Feel free to skim the rest of this file. The following changes are done:

- The code in the if block just has an indentation change.
- The code in the else block is removed.

File: stable/database/files/nuosm (hunk 1 above)
Comment (adriansuarez) [reply stable/database/files/nuosm:511]: Clicking the "Diff view" settings and selecting "Hide whitespace" helps.

**Review by adriansuarez — APPROVED** [review]
Excellent!

**Review by bkelly-ndb — APPROVED** [review]

**Pull request merged**